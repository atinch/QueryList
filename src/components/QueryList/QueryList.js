import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Table, Button } from 'semantic-ui-react';

export class QueryList extends React.Component {
 
  render() {
    return (
      <div>
        <Link to='/queries/new' align='left'><Button >New Query</Button></Link>
        <Table>
          <Table.Header>
            <Table.Row>
              <Table.HeaderCell width='2'>Name</Table.HeaderCell>
              <Table.HeaderCell>Query</Table.HeaderCell>
            </Table.Row>
          </Table.Header>

          <Table.Body>
            {this.props.queries.map(({ id, firstName, surname, query }) => (
              <Table.Row key={id}>
                <Table.Cell>{firstName} {surname}</Table.Cell>
                <Table.Cell>{query}</Table.Cell>
              </Table.Row>
            ))}
          </Table.Body>
        </Table>
      </div>
    )
  }
}


const mapStateToProps = (state) => ({
  queries: state.queries
});


export default connect(
  mapStateToProps
)(QueryList);
