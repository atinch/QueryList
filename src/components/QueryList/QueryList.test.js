import React from 'react';
import { QueryList } from './QueryList';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme';
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { Table, Button } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

configure({ adapter: new Adapter() });

const queries = [
  {
    id: 122432,
    firstName: 'Atinc',
    surname: 'Erguven',
    email: 'Atinc@gmail.com',
    query: 'Lorem'
  },
]

describe('QueryList', () => {

  const wrapper = shallow(<QueryList queries={queries} />);

  console.log(wrapper.debug());
  it('should assign correct header', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('should render a Button', () => {
   expect(wrapper.find(Button)).toHaveLength(1);
  });

  it('should render a Table', () => {
    expect(wrapper.find(Table)).toHaveLength(1);
   });

  it('should render Link correctly', () => {
    const link = wrapper.find(Link)
    expect(link).toHaveLength(1);
    expect(link.prop('to')).toBe('/queries/new')
  });
});


