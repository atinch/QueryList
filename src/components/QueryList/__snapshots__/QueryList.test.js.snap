// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryList should assign correct header 1`] = `
<div>
  <Link
    align="left"
    to="/queries/new"
  >
    <Button
      as="button"
    >
      New Query
    </Button>
  </Link>
  <Table
    as="table"
  >
    <TableHeader
      as="thead"
    >
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableHeaderCell
          as="th"
          width="2"
        >
          Name
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
        >
          Query
        </TableHeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody"
    >
      <TableRow
        as="tr"
        cellAs="td"
        key="122432"
      >
        <TableCell
          as="td"
        >
          Atinc
           
          Erguven
        </TableCell>
        <TableCell
          as="td"
        >
          Lorem
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</div>
`;
